From a84caf772ae110539703c8f998a12e3a8cf4d1ec Mon Sep 17 00:00:00 2001
From: Zahari Doychev <zdoychev@maxlinear.com>
Date: Mon, 15 Feb 2021 14:11:26 +0100
Subject: [PATCH] PONRTSYS-8526: datapath: pon_qos: use cookie for CPU queue
 map

Use cookie value instead of hardcoded values for the CPU queue to TC
mappings.
---
 drivers/net/datapath/pon_qos/pon_qos_tc_qos.c | 24 +-----------------------
 1 file changed, 1 insertion(+), 23 deletions(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c b/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c
index 43b65f9d312b..43670c8aa93d 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c
@@ -862,9 +862,6 @@ static int pon_qos_config_cpu_port(struct pon_qos_q_data *qid,
 {
 	int ret, i;
 
-	/* only 8 classes for CPU, starting from TC=8 */
-	subif += 8;
-
 	ret = pon_qos_set_qmap(qid, port, subif, tc_cookie, en);
 	if (ret != 0)
 		return ret;
@@ -874,16 +871,6 @@ static int pon_qos_config_cpu_port(struct pon_qos_q_data *qid,
 	else
 		atomic_inc(&qid->ref_cnt);
 
-	if (subif != 15)
-		return 0;
-
-	/* map class [0..7] and 15 to low prio queue */
-	for (i = 0; i < 7; i++) {
-		ret = pon_qos_set_qmap(qid, port, i, tc_cookie, en);
-		if (ret != 0)
-			pr_err("%s: cpu port config failed\n", __func__);
-	}
-
 	return ret;
 }
 
@@ -988,16 +975,7 @@ int pon_qos_update_qmap(struct net_device *dev,
 
 	/* TODO: DPM uses port zero currently as the CPU port */
 	if (pon_qos_is_cpu_port(sch->port)) {
-		/* CPU TC mappings are based on the class id handle for now i.e.
-		 * queue with classid X:Y is mapped tc = (Y-1) + 8.
-		 * Later this can be improved by using tc-cookie.
-		 */
-
-		tc = TC_H_MIN(q_tc->handle) - 1;
-		if (tc < 0 || tc > PON_QOS_8021P_HIGHEST_PRIO)
-			return -EINVAL;
-
-		ret = pon_qos_config_cpu_port(qid, sch->port, tc,
+		ret = pon_qos_config_cpu_port(qid, sch->port, 0,
 					      q_tc->tc_cookie, en);
 	} else {
 		ret = pon_qos_map_qid(qid, sch, tc, q_tc->tc_cookie, en);
